Code/Player/HarvestPlayer.cs

```diff
@@ -0,0 +1,149 @@
+using System.Diagnostics;
+using CatHarvest;
+using CatHarvest.UI;
+using CatHarvest.Util.Particles;
+using Sandbox;
+using Sandbox.Entities;
+using Sandbox.Services;
+
+public sealed class HarvestPlayer : Component
+{
+	public int CatsUprooted { get; set; } = 0;
+	public int CatsHarvested { get; set; } = 0;
+	[Property] public bool OpenInventory { get; set; } = false;
+	[Property] public ShrimpleCharacterController.ShrimpleCharacterController _playerController;
+	public bool CloseInstructions { get; set; } = false;
+	public bool HasCat { get; set; } = false;
+	public enum PopupType
+	{
+		None,
+		Uproot,
+		SecretPickUp
+	}
+	
+	public PopupType Popup { get; set; } = PopupType.None;
+	
+	public CameraComponent OverrideCamera { get; set; } = null;
+
+	public Vector3 InputDirection { get; set; }
+	public Angles ViewAngles { get; set; }
+
+	public SkinnedModelRenderer ViewModel { get; set; }
+
+	protected override void OnStart()
+	{
+		Tags.Add("Player");
+	}
+
+	protected override void OnUpdate()
+	{
+
+	}
+
+	TimeSince lastStep = 0f;
+	TimeSince autoClose = 0f;
+	protected override void OnFixedUpdate()
+	{
+		if ( IsProxy ) return;
+		if ( !GameObject.IsValid() ) return;
+		if ( HarvestGame.The.Finishing ) return;
+		OpenInventory = Input.Down( "Score" );
+		if ( autoClose >= 15f || OpenInventory)
+			CloseInstructions = true;
+		
+		var eyeTraceSetup = Scene.Trace.Ray( new Ray( Scene.Camera.WorldPosition, Scene.Camera.WorldTransform.Forward ), 75f )
+			.Size( new Vector3( 20f, 20f, 20f ) )
+			.IgnoreGameObjectHierarchy(this.GameObject.Root)
+			.WithTag( "Cat" );
+		var eyeTrace = eyeTraceSetup.Run();
+		var lookingAtCat = eyeTrace.Hit && eyeTrace.GameObject.Tags.Has( "cat" );
+		if ( lookingAtCat && !HasCat )
+		{
+			var cat = eyeTrace.GameObject;
+			if ( cat.GetComponent<WalkingCat>().IsValid() && cat.GetComponent<WalkingCat>().IsSecret )
+			{
+				Popup = PopupType.SecretPickUp;
+
+				if ( Input.Pressed( "use" ) )
+				{
+					cat.Destroy();
+
+					GetComponentInChildren<SkinnedModelRenderer>().Set( "wiwi", true );
+					HarvestGame.EndGame( this, 0, true );
+				}
+			}
+			else
+			{
+				if ( cat.GetComponent<WalkingCat>().IsValid() ) return;
+				Popup = PopupType.Uproot;
+
+				if ( Input.Pressed( "use" ) )
+				{
+					Sound.Play( $"meow{Game.Random.Int( 10 )}", cat.WorldPosition );
```

Comment:
i think you can combine all the meows into one sound file and then use s&box's built in random sound player mode